diff -ruN u/defaults/main.yml v/defaults/main.yml
--- u/defaults/main.yml	2018-04-06 09:47:50.663311051 +0200
+++ v/defaults/main.yml	2018-04-09 09:26:23.312742521 +0200
@@ -313,6 +313,8 @@
   - trove_dashboard
   - designate_dashboard
   - octavia_dashboard
+  - python-monascaclient
+  - monasca-ui
 
 # This variable is used to install additional pip packages
 # that could be needed for additional dashboards
@@ -341,3 +343,7 @@
 horizon_websso_initial_choice: "credentials"
 
 horizon_panels: "{{ _horizon_panels }}"
+
+#Monasca
+horizon_enable_monasca_ui: True
+horizon_grafana_url: "https://{{ external_lb_vip_address }}:3000"
diff -ruN u/tasks/horizon_post_install.yml v/tasks/horizon_post_install.yml
--- u/tasks/horizon_post_install.yml	2018-04-06 09:47:20.038962749 +0200
+++ v/tasks/horizon_post_install.yml	2018-01-11 12:14:27.000000000 +0100
@@ -103,3 +103,16 @@
   register: async_compress_static_files
   async: 600
   poll: 0
+
+- name: Enable the monasca-ui-dashboard Horizon panel
+  file:
+    src: "{{ horizon_lib_dir }}/monitoring/enabled/_50_admin_add_monitoring_panel.py"
+    dest: "{{ horizon_lib_dir }}/openstack_dashboard/local/enabled/_50_admin_add_monitoring_panel.py"
+    state: "{{ horizon_enable_monasca_ui | ternary('link', 'absent') }}"
+  notify: Restart apache2
+
+- name: Drop monasca-ui Config(s)
+  template:
+    src: "monasca-ui/local_settings.py.j2"
+    dest: "{{ horizon_lib_dir }}/monitoring/config/local_settings.py"
+  notify: Restart apache2
diff -ruN u/templates/monasca-ui/local_settings.py.j2 v/templates/monasca-ui/local_settings.py.j2
--- u/templates/monasca-ui/local_settings.py.j2	1970-01-01 01:00:00.000000000 +0100
+++ v/templates/monasca-ui/local_settings.py.j2	2017-12-05 06:32:37.000000000 +0100
@@ -0,0 +1,86 @@
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License. You may obtain
+#  a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+# Service group names (global across all projects):
+MONITORING_SERVICES_GROUPS = [
+    {'name': _('OpenStack Services'), 'groupBy': 'service'},
+    {'name': _('Servers'), 'groupBy': 'hostname'}
+]
+
+# Services being monitored
+MONITORING_SERVICES = getattr(
+    settings,
+    'MONITORING_SERVICES_GROUPS',
+    MONITORING_SERVICES_GROUPS
+)
+
+#
+# Per project service groups.  If in this form,
+# '*' will be applied to all projects not explicitly listed.
+#
+# Note the above form (flat) is supported for backward compatibility.
+#
+# MONITORING_SERVICES_GROUPS = [
+#    {'admin': [
+#         {'name': _('OpenStack Services'), 'groupBy': 'service'},
+#         {'name': _('Servers'), 'groupBy': 'hostname'}]},
+#    {'*': [
+#         {'name': _('Services'), 'groupBy': 'service'},
+#         {'name': _('Instances'), 'groupBy': 'hostname'}]},
+# ]
+
+MONITORING_SERVICE_TYPE = getattr(
+    settings, 'MONITORING_SERVICE_TYPE', 'monitoring'
+)
+
+# Grafana button titles/file names (global across all projects):
+GRAFANA_LINKS = [
+
+{% if (groups['ceph_all'] is defined) and (groups['ceph_all'] | length > 0) %}
+    {'title': 'Ceph Dashboard', 'path': 'ceph-dashboard'},
+    {'title': 'Ceph OSDs Dashboard', 'path': 'ceph-osds-dashboard'},
+    {'title': 'Ceph Pools Dashboard', 'path': 'ceph-pools-dashboard'},
+{% endif %}
+    {'title': 'Containers Dashboard', 'path': 'containers-dashboard'},
+    {'title': 'Hosts Dashboard', 'path': 'hosts-dashboard'},
+    {'title': 'Infra Dashboard', 'path': 'infra-dashboard'},
+    {'title': 'Instances Dashboard', 'path': 'instances-dashboard'}
+]
+DASHBOARDS = getattr(settings, 'GRAFANA_LINKS', GRAFANA_LINKS)
+
+#
+# Horizon will link to the grafana home page when using Grafana2.
+# For any Grafana version additional links to specific dashboards can be
+# created in two formats.
+# Flat:
+# GRAFANA_LINKS = [ {'title': _('Dashboard'), 'path': 'openstack'} ]
+#
+# Per project: '*' will be applied to all projects not explicitly listed.
+# GRAFANA_LINKS = [
+#    {'admin': [
+#        {'title': _('Dashboard'), 'path': 'openstack'}]},
+#    {'*': [
+#        {'title': _('OpenStack Dashboard'), 'path': 'project'}]}
+# ]
+
+GRAFANA_URL = { '{{ horizon_service_region }}': '{{ horizon_grafana_url }}', }
+
+ENABLE_KIBANA_BUTTON = getattr(settings, 'ENABLE_KIBANA_BUTTON', False)
+KIBANA_POLICY_RULE = getattr(settings, 'KIBANA_POLICY_RULE', 'admin_required')
+KIBANA_POLICY_SCOPE = getattr(settings, 'KIBANA_POLICY_SCOPE', 'identity')
+KIBANA_HOST = getattr(settings, 'KIBANA_HOST', 'http://192.168.10.4:5601/')
+
+OPENSTACK_SSL_NO_VERIFY = getattr(settings, 'OPENSTACK_SSL_NO_VERIFY', False)
+OPENSTACK_SSL_CACERT = getattr(settings, 'OPENSTACK_SSL_CACERT', None)
